app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -259,10 +259,31 @@ public void muteTopic(Message message) {
 
     @Override
     public void recyclerViewScrolled() {
-        if (chatBox.getVisibility() == View.VISIBLE && !isTextFieldFocused) {
+
+        if(chatBox.getVisibility()== View.VISIBLE && !isCurrentModeStream()){
+            //check if messageEt is empty or not
+            if(messageEt.getText().toString().equals(""))
+            {
+                displayChatBox(false);
+                displayFAB(true);
+
+            }
+        }
+
+        else if(chatBox.getVisibility()==View.VISIBLE && isCurrentModeStream()){
+            //check if messageEt is empty or not
```

Comment:
A common piece of advice on code comments is that we should not merely say what the code is doing, but why we have made that choice -- giving context about our design choices for future people reading the code who might not understand what bug we are trying to solve, or what problem we're trying to work around. In this case you might choose to leave a code comment on why it matters whether messageEt is empty. But I'm less familiar with the Android codebase here so I may be wrong in suggesting that it's useful or necessary to do that here -- I'll leave that to @kunall17's judgment.